### PR TITLE
sixlowpan: initial import of 6LoWPAN-ND submodule

### PR DESCRIPTION
**Pull request by authmillenon**
This PR imports the ICMPv6 message types and NDP option types introduced by 6LoWPAN-ND ([RFC 6775](https://tools.ietf.org/html/rfc6775)).

Updated files:
 * `net/icmpv6.h`: `ICMPV6_DAR` and `ICMPV6_DAC` type number definitions for duplicate address request and confirmation messages.
 * `net/ndp.h`: `NDP_OPT_AR`, `NDP_OPT_6CTX`, and `NDP_OPT_ABR` type number definitions for the new option types.

New file:
* `net/sixlowpan/nd.h` containing the type defintitions and helper functions for the 6LoWPAN-ND messages and options.

### CODE REVIEW

File: sys/include/net/ndp.h [hunk 1]
```diff
@@ -69,6 +69,9 @@ extern "C" {
 #define NDP_OPT_PI                  (3)     /**< prefix information option */
 #define NDP_OPT_RH                  (4)     /**< redirected option */
 #define NDP_OPT_MTU                 (5)     /**< MTU option */
+#define NDP_OPT_AR                  (33)    /**< address registration option */
+#define NDP_OPT_6CTX                (34)    /**< 6LoWPAN context option */
+#define NDP_OPT_ABR                 (36)    /**< authoritative border router option */
```
Comment (cgundogan) [inline sys/include/net/ndp.h]: according to [rfc6775#section-4.3](https://tools.ietf.org/html/rfc6775#section-4.3), this value must be (35)

Comment (authmillenon) [inline sys/include/net/ndp.h]: Oh, you're right. Must have been a typo. Fixed!

**Comment by cgundogan** [comment]
Could you briefly elaborate on why you labeld this as `API CHANGE`? I couldn't find anything regarding this in the diff (only line additions, no deletions or modifications)

**Comment by authmillenon** [comment]
IMHO API extension is an API change, too. You may remove it, if you are of a different opinion ;-)

**Comment by cgundogan** [comment]
I was just curious about it - it's fine this way (: ACK from my side. Please squash.

**Comment by authmillenon** [comment]
Squashed

**Comment by cgundogan** [comment]
Why again have been >90% of all tests skipped?

**Comment by authmillenon** [comment]
Because those tests don't use the files changed here

**Pull request merged**